### ISSUE TITLE
False positive ParadoxicalCondition with loop

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#202]
False positive ParadoxicalCondition reproduction case is below (unrelated lines were removed)

Seen with vimeo/psalm 0.3.53 (and master?)

```php
<?php

/** @return bool[] */
function get_values() {
    return [true, false];
}

function paradox2() : void {
    $condition = rand() % 2 > 0;
    if (!$condition) {
        foreach (get_values() as $value) {
            if ($condition) { // false positive warning
                echo "previous was true\n";
            }
            $condition = $value;  // it can change within the loop
        }
    }   
}
```

**Comment by muglug** [comment]
Thanks for the great test case – should be fixed now
